lib/vtls/nss.c

```diff
@@ -1837,8 +1864,15 @@ static CURLcode nss_setup_connect(struct connectdata *conn, int sockindex)
                                   &sslver_supported) != SECSuccess)
     goto error;
   if(sslver_supported.max < sslver.max && sslver_supported.max >= sslver.min) {
-    infof(data, "Falling back (from %d) to max supported SSL version (%d)\n",
-                sslver.max, sslver_supported.max);
+    sslver_req_str = nss_sslver_to_name(sslver.max);
+    sslver_supp_str = nss_sslver_to_name(sslver_supported.max);
+    if(sslver_req_str && sslver_supp_str)
+      infof(data, "Falling back from %s to max supported SSL version (%s)\n",
+                  sslver_req_str, sslver_supp_str);
+    if(sslver_req_str)
```

Comment:
I would optimize out the `if` statements before the calls to `free()`.  The code already relies on `free(NULL)` being safe.

Reply:
Fix that too.